### ISSUE TITLE
Display mobile menu at XS and SM screen sizes

### ISSUE THREAD

**Issue by danfox01** [baltimorecounty/BCPL-assets#592]
**Background**

Currently, the hamburger icon and mobile menu appear only at XS screen sizes, which traditionally comprised portrait and landscape views for most mobile phones. Newer phones (included all 2018 iPhones) are so large that they will actually trigger the tablet view when viewed in landscape. This is fine for content, but it's a bad experience for menus, which are displayed differently at these two sizes.

**Requirements**

- Update navigation so that the mobile menu is displayed at both XS and SM sizes (up to 992px).
- All non-menu content should be displayed as usual for the SM breakpoint.
- In other words, the display change should apply only to the nav at XS sizes -- all other elements should stay tablet-like.
- The updated SM nav style needs to be ported into Polaris as well.

Work in progress from @martypowell : https://github.com/baltimorecounty/BCPL-assets/tree/xs-navigation-udpates
